dropshot_endpoint/src/lib.rs

```diff
@@ -286,6 +290,33 @@ fn do_endpoint(
         })
         .collect::<Vec<_>>();
 
+    // We want to construct a function that will call the user's endpoint, so
+    // we can check the future it returns for bounds that otherwise produce
+    // inscrutable error messages (like returning a non-`Send` future). We
+    // produce a wrapper function that takes all the same argument types,
+    // which requires building up a list of argument names.
+    let arg_names = (0..arg_types.len())
+        .map(|i| {
+            let argname = format_ident!("arg{}", i);
+            quote! { #argname }
+        })
+        .collect::<Vec<_>>();
+    let impl_checks = if !arg_is_receiver {
+        quote! {
+            const _: fn() = || {
+                fn future_endpoint_must_be_send<T: ::std::marker::Send>(_t: T) {}
+                fn check_future_bounds(#( #arg_names: #arg_types ),*) {
+                    #item
```

Comment:
This is the implementation wart I referred to in the PR comment; it's a full duplicate of the endpoint definition. At first I tried moving this into an `impl` block on the struct named `#name`, but that has adverse effects on the error message case for endpoints taking `self`, so went with this instead. Open to other ideas!

Reply:
Fixed in [8568fba](https://github.com/oxidecomputer/dropshot/pull/293/commits/8568fba136488bf74d08a8f782b50ba91a022448); thanks @ahl for the suggestion